app/src/ts.app.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * Broadcast Message to one or more apps.
+ * @param {array|string} appIds List of apps to receive the message, supports glob
+ * Examples:
+ * * ['Tradeshift.Developer', 'Tradeshift.DeveloperDemo'] (matches Tradeshift.Developer and Tradeshift.DeveloperDemo)
+ * * 'Tradeshift.Developer' (matches Tradeshift.Developer)
+ * * '*' (matches any app)
+ * * 'Tradeshift.*' (matches all Tradeshift apps)
+ * * 'Tradeshift.??Y' (matches Tradeshift.Buy, Tradeshift.Pay, etc.)
+ * @param {string} key The key/subject of the event
+ * @param {object} data Data to be sent with the event, has to support the structured clone algorithm
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm
+ */
+export function broadcast(appIds, key, data) {
+	const content = stringify(appIds, key, data);
+	parent.postMessage(content, '*');
+}
+
+/**
+ * Subscribe to messages from one or more apps.
+ * @param {array|string} appIds List of apps to receive the message, supports glob
+ * @return {Listener} Listener object
+ */
+export function subscribe(appIds) {
+	return new Listener(appIds);
+}
+
+const BROADCAST_PREFIX = 'app-broadcast:';
+
+/**
+ * Message listener for one or more apps.
+ */
+class Listener {
+	/**
+	 * Create the listener.
+	 * @param {array|string} appIds 
+	 */
+	constructor(appIds) {
+		this.appIds = appIds;
+		this.events = new Map();
+	}
```

Comment:
I always enjoy the babel plugin `babel-plugin-transform-class-properties` (or something like that) that allows you to declare instance properties like this:

```javascript
class Listener {
   appIds = appIds;
   events = new Map();
}
```
— which is much easier to write JSDoc comments for. This plugin will become default in Babel 7, by the way, so that you won't need to install it separately.